scheduler/scheduler-server/src/test/java/functionaltests/job/log/TestJobServerLogs.java

```diff
@@ -160,7 +160,22 @@ public void checkRemoval(JobId jobId) throws Exception {
 
         checkJobAndTaskLogFiles(jobId, tasks, true);
 
+        // Before it was only job removal but it was bug prone (for pendingJob only)
+        // because the following sequence of the events could happen:
+        // 1. SchedulerMethodImpl main loop retrieves info about this forever pending job
+        // 2. we call removeJob
+        // 3. SchedulerMethodImpl main loop asks for RM::getRMNodes.
+        //
+        // Third action would print "[SelectionManager.doSelectNodes] "scheduler"..." to the task log
+        // which would actually re-created removed log file which would cause an error
+        // in the last line of this method
+        // that is why kill the job and after wait 5s to make sure that
+        // iteration of SchedulerMethodImpl main loop is finished,
+        // and finally, we remove job with its log files.
+        schedulerHelper.killJob(jobId.value());
+        Thread.sleep(5000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

Reply:
So we can remove this sleep now or do we still need it?

Reply:
The idea is to wait until iteration of main Scheduler is finished.
So that is why I sleep 5s.

Reply:
ok